lib/daylite/contact.rb

```diff
@@ -0,0 +1,28 @@
+class Daylite::Contact < Daylite::Base
+  belongs_to :category, foreign_key: "categoryid"
+  belongs_to :owner, class_name: "User", foreign_key: "assignedtoid"
+
+  belongs_to :details, class_name: "LargeData", foreign_key: "detailsid"
+
+  has_one :user, foreign_key: "contactid"
+
+  has_many :contacttaskjoins, foreign_key: "contactid"
```

Comment:
Those names do not follow usual rails/ruby convention of underscoring/casing separate words in a compound name. I think it's important to keep general style of ruby code in sync and name like `contact_task_joins` should be used, the `ContactTaskJoin` can override table name to be `contacttaskjoins` but hide this ugly fact from the ruby code :)

Reply:
Reflecting on the fact that currently released version already uses those non-separated names for join tables and models this could be kept as is, but maybe we should apply underscores to associations which will be most likely used by clients, f.e. `roletypes -> role_types`. @kitebuggy, what do you think?

Reply:
I would suggest we move to the proper Ruby convention, but include a deprecation warning for the existing (old, Daylite-inspired) names.  It is unlikely that users will have accessed the join tables directly for most use cases, but on occasion I have done so, e.g. to clear out old 'deleted' records from the join tables.